charts/internal/machineclass/templates/machineclass.yaml

```diff
@@ -18,9 +18,14 @@ kind: MachineClass
 metadata:
   name: {{ $machineClass.name }}
   namespace: {{ $.Release.Namespace }}
-{{- if $machineClass.labels }}
+{{- if or $machineClass.operatingSystem $machineClass.labels }}
```

Comment:
I actually think this condition is not necessary (and same for the {{end}} at l. 29. Can you give that a shot ?

Reply:
yes, it also works. new commit 0f346ec0 @kon-angelo